### ISSUE TITLE
Add passby_press to basebutton

### ISSUE THREAD

**Issue by likeich** [godotengine/godot-proposals#3030]
### Describe the project you are working on

An Android launcher that is ui-heavy. (For IOS people-an android launcher is basically an app that you can set to override your home screen)

### Describe the problem or limitation you are having in your project

I'm trying to make a specific piece of ui that has proven to be difficult because of the texture button not supporting passby presses. Specifically, I'm trying to make something similar to this character slider on the right in the screenshot below. You can drag your finger over the letters and the launcher will bring you to apps with that letter. This has proven difficult because there's no built in support for control buttons to press a button when the mouse/finger slides over it.

![Screenshot_20210722-103858_Niagara Launcher](https://user-images.githubusercontent.com/57882701/126658079-540faec3-ff67-4227-b8d1-17d45b22ad65.jpg)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add the passby_press property from the touchscreen button to the base button. This would make it so I could replicate this feature really easily.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Basically copy and paste the functionality of passby_press to the base button.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I've tried and it's difficult/hacky to do. Currently the best method I've found is making one out of touchscreen buttons. But the problem with this is I lose the automatic scaling and all the other benefits of control nodes.

### Is there a reason why this should be core and not an add-on in the asset library?

It concerns GUI functionality, which is core.

**Comment by Calinou** [comment]
> Basically copy and paste the functionality of passby_press to the base button.

It's likely more involved than just copy-pasting, as TouchScreenButton inherits from Node2D whereas BaseButton inherits from Control.

**Comment by likeich** [comment]
I definitely agree that updating the touchscreen button is long overdue. That needs a proposal by itself.

I want to clarify that I'm not looking for pressing the button when it's hovered, passby press works when a click happens outside of the button and then drags over the button. This is different from hovering which implies that no click occurred

**Comment by Calinou** [comment]
> I want to clarify that I'm not looking for pressing the button when it's hovered, passby press works when a click happens outside of the button and then drags over the button. This is different from hovering which implies that no click occurred

In this case, this sounds like a good feature to add, even for desktop platforms. There are many non-game applications that could benefit from this pattern

**Comment by Kazox61** [comment]
> I want to clarify that I'm not looking for pressing the button when it's hovered, passby press works when a click happens outside of the button and then drags over the button. This is different from hovering which implies that no click occurred

Can this be achieved by the `hit_margin` from https://github.com/godotengine/godot/pull/116763.

**Comment by StockerGaming** [comment]
> Can this be achieved by the `hit_margin` from [godotengine/godot#116763](https://github.com/godotengine/godot/pull/116763).

Just to put that info into this thread here too, this does not seem to be the case, as stated in [the discussion that mentioned this issue](https://github.com/godotengine/godot/pull/120693#discussion_r3499986950)
